pkg/comp-functions/functions/vshnpostgres/pgqexporter_workaround.go

```diff
@@ -0,0 +1,566 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	xkubev1 "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	"github.com/crossplane/function-sdk-go/proto/v1beta1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+func PgExporterWorkaround(ctx context.Context, svc *runtime.ServiceRuntime) *v1beta1.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite: %w", err))
+	}
+
+	log.Info("Making sure the cluster exposed connection details")
+	obj := &xkubev1.Object{}
+	err = svc.GetObservedComposedResource(obj, "cluster")
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot get the sgcluster object: %s", err))
+	}
+
+	// get configmap
+	configMap := &corev1.ConfigMap{}
+	// create k8s client inside cluster
+	config, err := rest.InClusterConfig()
```

Comment:
Please use the kube objects here. This adhoc k8s client will bite us in the ass if we ever want to do something on remote clusters.